include/ringmesh/mesh/mesh_base.h

```diff
@@ -0,0 +1,136 @@
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in the
+ *       documentation and/or other materials provided with the distribution.
+ *     * Neither the name of ASGA nor the
+ *       names of its contributors may be used to endorse or promote products
+ *       derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL ASGA BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+ * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *     http://www.ring-team.org
+ *
+ *     RING Project
+ *     Ecole Nationale Superieure de Geologie - GeoRessources
+ *     2 Rue du Doyen Marcel Roubault - TSA 70605
+ *     54518 VANDOEUVRE-LES-NANCY
+ *     FRANCE
+ */
+
+#pragma once
+
+#include <ringmesh/mesh/common.h>
+
+#include <algorithm>
+#include <memory>
+
+#include <ringmesh/basic/factory.h>
+#include <ringmesh/basic/nn_search.h>
+
+#include <ringmesh/mesh/mesh_aabb.h>
+
+namespace GEO
+{
+    class AttributesManager;
+} // namespace GEO
+
+namespace RINGMesh
+{
+    FORWARD_DECLARATION_DIMENSION_CLASS( GeoModel );
+    FORWARD_DECLARATION_DIMENSION_CLASS( MeshBaseBuilder );
+    FORWARD_DECLARATION_DIMENSION_CLASS( PointSetMeshBuilder );
+    FORWARD_DECLARATION_DIMENSION_CLASS( LineMeshBuilder );
+    FORWARD_DECLARATION_DIMENSION_CLASS( SurfaceMeshBuilder );
+    FORWARD_DECLARATION_DIMENSION_CLASS( VolumeMeshBuilder );
+    FORWARD_DECLARATION_DIMENSION_CLASS( SurfaceMesh );
+
+    struct EdgeLocalVertex;
+    struct ElementLocalVertex;
+    struct PolygonLocalEdge;
+    struct CellLocalFacet;
+} // namespace RINGMesh
+
+namespace RINGMesh
+{
+    /*!
+     * class base class for encapsulating Mesh structure
+     * @brief encapsulate adimensional mesh functionalities in order to provide
+     * an API
+     * on which we base the RINGMesh algorithms
+     * @note For now, we encapsulate the GEO::Mesh class.
+     */
+    // BEGINING OF MESHBASE DEFINITION

```

Comment:
Do not add comment like this. Try to prefer doxygen comment or change your code so it does not need any comment ;)